### PR TITLE
Fixing Compiler Warning in TrilinosApp

### PR DESCRIPTION
**Pull request by philbucher**
Fixed a compiler warning and changed increment type (i++ => ++i)

### CODE REVIEW

File: applications/trilinos_application/trilinos_space.h [hunk 1]
```diff
@@ -564,7 +564,7 @@ class TrilinosSpace
         rGraph.RowMap().MyGlobalElements(MyGlobalElements);
         
         
-        for(IndexType i=0; i< NumMyRows; i++)
+        for(IndexType i = 0; i < static_cast<IndexType>(NumMyRows); ++i)
```
Comment (jcotela) [inline applications/trilinos_application/trilinos_space.h]: I think the cast is unnecessary here. I would just declare NumMyRows as an IndexType above.
```
IndexType NumMyRows = ....
```
NumMyRows is used to allocate an array in addition to here, so if it is a negative number we have a problem there too.

**Review by jcotela — APPROVED** [review]
I approve this PR, although I would recommend doing the change I'm suggesting.

File: applications/trilinos_application/trilinos_space.h (hunk 1 above)
Comment (philbucher) [reply applications/trilinos_application/trilinos_space.h]: I wasn't sure abt this, will change it

**Comment by jcotela** [comment]
Thanks for the change. Go ahead and merge as soon as compilation test passes

**Comment by roigcarlo** [comment]
Also, I don't see any reason to start changing post-increment to pre-increments. I think we discussed this in other PR but anyway, just an observation :+1:

**Comment by philbucher** [comment]
@roigcarlo I only did it in this case bcs I changed that line anyway.
Usually I wouldn't change this, thx for pointing it out

**Comment by jcotela** [comment]
@roigcarlo I agree, pre-increments are good in some cases, but there is no need to use them systematically. Here it makes no difference.

**Pull request merged**